include/SDL3/SDL_video_capture.h

```diff
@@ -0,0 +1,367 @@
+
+/**
+ * Number of available formats for the device
+ *
+ * \param device opened video capture device
+ * \returns number of formats or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetVideoCaptureFormat
+ * \sa SDL_SetVideoCaptureSpec
+ */
+extern DECLSPEC int SDLCALL SDL_GetNumVideoCaptureFormats(SDL_VideoCaptureDevice *device);
+
+/**
+ * Get frame sizes of the device and the specified input format.
+ * The value can be used to fill SDL_VideoCaptureSpec structure.
+ *
+ * \param device opened video capture device
+ * \param format a format that can be used by the device (SDL_PixelFormatEnum)
+ * \param index framesize between 0 and num -1
+ * \param width output width
+ * \param height output height
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetNumVideoCaptureFrameSizes
+ */
+extern DECLSPEC int SDLCALL SDL_GetVideoCaptureFrameSize(SDL_VideoCaptureDevice *device, Uint32 format, int index, int *width, int *height);
+
+/**
+ * Number of different framesizes available for the device and pixel format.
+ *
+ * \param device opened video capture device
+ * \param format frame pixel format (SDL_PixelFormatEnum)
+ * \returns number of framesizes or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetVideoCaptureFrameSize
+ * \sa SDL_SetVideoCaptureSpec
+ */
+extern DECLSPEC int SDLCALL SDL_GetNumVideoCaptureFrameSizes(SDL_VideoCaptureDevice *device, Uint32 format);
+
+
+/**
+ * Get video capture status
+ *
+ * \param device opened video capture device
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_VideoCaptureStatus
+ */
+extern DECLSPEC SDL_VideoCaptureStatus SDLCALL SDL_GetVideoCaptureStatus(SDL_VideoCaptureDevice *device);
+
+/**
+ * Start video capture
+ *
+ * \param device opened video capture device
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_StopVideoCapture
+ */
+extern DECLSPEC int SDLCALL SDL_StartVideoCapture(SDL_VideoCaptureDevice *device);
+
+/**
+ * Acquire a frame.
+ * The frame is a memory pointer to the image data, whose size and format
+ * are given by the the obtained spec.
+ *
+ * Non blocking API. If there is a frame available, frame->num_planes is non 0.
+ * If frame->num_planes is 0 and returned code is 0, there is no frame at that time.
+ *
+ * After used, the frame should be released with SDL_ReleaseVideoCaptureFrame
```

Comment:
I'd say yes. because there several buffer available internally. 
but it may also depend on the hardware, which could prevent from acquiring a new frame it the first one haven't been release.

Reply:
edit: 
the generic layer, doesn't limit the number of buffer.  
it's the video capture back-end that can push as many frame as it want/can.
for linux, the backend is hard-coded at 8 buffers.
if the user doesn't release the 8 frame, the acquisition is stucked / lacking free buffers